Sources/KituraNet/HTTP/HTTPServerRequest.swift

```diff
@@ -41,18 +41,24 @@ public class HTTPServerRequest: ServerRequest {
 
     @available(*, deprecated, message: "URLComponents has a memory leak on linux as of swift 3.0.1. use 'urlURL' instead")
     public var urlComponents: URLComponents {
-        return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
+        if let _urlComponents = _urlComponents {
+            return _urlComponents
+        } else {
+            return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
```

Comment:
@nethraravindran We must cache it here as well, in the scenario where `urlComponents` was invoked before `urlURL`

Reply:
@pushkarnk Sure!